### PR TITLE
DDF-05318 Pass fetch method to source-item and call it on close

### PR DESCRIPTION
**Pull request by harrison-tarr**
#### What does this PR do?
Passes a callback to the models fetch method to the view, allowing the view to refresh itself
#### Select relevant component teams: 
@Schachte 
@djblue 

@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@adimka
@andrewkfiedler

#### How should this be tested?
Full build should be enough, it just adds framework extensibility options without an impl

#### What are the relevant tickets?
Fixes: https://github.com/codice/ddf/issues/5318

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/sources/container.tsx [hunk 1]
```diff
@@ -55,6 +55,7 @@ class SourcesSummaryContainer extends React.Component<Props, State> {
     return (
       <Sources
         sources={this.state.sources}
+        refreshSources={() => sources.fetch()}
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/sources/container.tsx]: Will sources always be defined? If it is, will they always have a fetch function?

Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/sources/container.tsx]: Looks like it's the imported backbone collection instance. Should be good to go. 👍

**Comment by harrison-tarr** [comment]
build now

**Comment by harrison-tarr** [comment]
build now

**Pull request merged**